src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,41 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+            return actor;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can not add actor: " + actor, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```

Comment:
Consider using try-with-resources for the `Session` object in the `add` method. This will ensure that the session is closed automatically, even if an exception occurs, and will help prevent resource leaks.